pkg/asset/manifests/openshift.go

```diff
@@ -120,8 +135,7 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 		"99_openshift-cluster-api_worker-user-data-secret.yaml": worker.UserDataSecretRaw,
 	}
 
-	switch platform {
-	case "aws", "openstack":
+	if cloudCreds != (cloudCredsSecretData{}) {
```

Comment:
maybe old way was more appropiate?

Reply:
I was looking to avoid having to add to the list of platforms here as well as at https://github.com/openshift/installer/blob/d7bf5b1aa05ce026b264b2fb8cddbc787c928a62/pkg/asset/manifests/cloudproviderconfig.go#L74-L101. Is this not an accepted pattern for checking whether a variable is a zero value?